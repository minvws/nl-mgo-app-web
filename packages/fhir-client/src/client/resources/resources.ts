import { type Bundle, type FhirVersion, type ResourceType } from '@minvws/mgo-fhir-types';
import { type Lossless } from '@minvws/mgo-mgo-utils';
import type { KyInstance, Options as KyOptions } from 'ky';
import type { FhirClientOptions } from '..';

export interface BaseResourcesRequest<
    V extends FhirVersion,
    T extends ResourceType<V> = ResourceType<V>,
> {
    resource: T;
    id?: never;
}

export interface ResourcesRequest<
    V extends FhirVersion,
    T extends ResourceType<V> = ResourceType<V>,
> extends BaseResourcesRequest<V, T> {
    $lastn?: never;
}

export interface ObservationResourcesRequest<
    V extends FhirVersion,
    T extends ResourceType<V> = 'Observation',
> extends BaseResourcesRequest<V, T> {
    $lastn?: boolean;
}

export type ResourcesResponse<
    V extends FhirVersion,
    T extends ResourceType<V>,
    Response = Lossless<Bundle<V, T>>,
> = Response;

export function setupResources<V extends FhirVersion>(
    instance: KyInstance,
    _options: FhirClientOptions<V>
) {
    return {
        getResources: <
            Request extends ResourcesRequest<V> | ObservationResourcesRequest<V>,
            Response = ResourcesResponse<V, Request['resource']>,
        >(
            request: Request,
            options: KyOptions = {}
        ) => {
            const { resource, $lastn } = request as ObservationResourcesRequest<V>;
            return instance.get<Response>(`${resource}${$lastn ? '/$lastn' : ''}`, options);
        },
    };
}
