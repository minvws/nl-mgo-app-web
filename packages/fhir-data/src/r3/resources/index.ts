/**
 * Configurations for parsing FHIR resources and creating UI schemas.
 * Add new R3 resource configurations here.
 *
 * Notes:
 *   - Make sure you are ONLY exporting `ResourceConfig` objects from this file.
 *   - Don't forget to also export the ResourceType from the `~/api/types` file to make it available for the mobile apps.
 */

export * from './gpDiagnosticResult/gpDiagnosticResult';
export * from './gpEncounter/gpEncounter';
export * from './gpEncounterReport/gpEncounterReport';
export * from './gpJournalEntry/gpJournalEntry';
export * from './gpLaboratoryResult/gpLaboratoryResult';

export * from './nlCoreEpisodeofcare/nlCoreEpisodeofcare';
export * from './nlCoreOrganization/nlCoreOrganization';
export * from './nlCorePatient/nlCorePatient';
export * from './nlCorePractitioner/nlCorePractitioner';
export * from './nlCorePractitionerRole/nlCorePractitionerRole';

export * from './eAfspraakAppointment/eAfspraakAppointment';
export { nlCoreObservation, type NlCoreObservation } from './nlCoreObservation/nlCoreObservation';
export * from './zibAdministrationAgreement/zibAdministrationAgreement';
export * from './zibAdvanceDirective/zibAdvanceDirective';
export * from './zibAlcoholUse/zibAlcoholUse';
export * from './zibAlert/zibAlert';
export * from './zibAllergyIntolerance/zibAllergyIntolerance';
export * from './zibBloodPressure/zibBloodPressure';
export * from './zibBodyHeight/zibBodyHeight';
export * from './zibBodyWeight/zibBodyWeight';
export * from './zibDrugUse/zibDrugUse';
export { zibEncounter, type ZibEncounter } from './zibEncounter/zibEncounter';
export * from './zibFunctionalOrMentalStatus/zibFunctionalOrMentalStatus';
export {
    zibGeneralMeasurement,
    type ZibGeneralMeasurement,
} from './zibGeneralMeasurement/zibGeneralMeasurement';
export {
    zibLaboratoryTestResultObservation,
    type ZibLaboratoryTestResultObservation,
} from './zibLaboratoryTestResultObservation/zibLaboratoryTestResultObservation';
export * from './zibLaboratoryTestResultSpecimen/zibLaboratoryTestResultSpecimen';
export * from './zibLaboratoryTestResultSpecimenIsolate/zibLaboratoryTestResultSpecimenIsolate';
export * from './zibLaboratoryTestResultSubstance/zibLaboratoryTestResultSubstance';
export * from './zibLivingSituation/zibLivingSituation';
export * from './zibMedicalDevice/zibMedicalDevice';
export * from './zibMedicalDeviceProduct/zibMedicalDeviceProduct';
export * from './zibMedicalDeviceRequest/zibMedicalDeviceRequest';
export * from './zibMedicationAgreement/zibMedicationAgreement';
export * from './zibMedicationUse/zibMedicationUse';
export * from './zibNutritionAdvice/zibNutritionAdvice';
export * from './zibPayer/zibPayer';
export * from './zibProblem/zibProblem';
export * from './zibProcedure/zibProcedure';
export * from './zibProcedureRequest/zibProcedureRequest';
export * from './zibProduct/zibProduct';
export * from './zibTobaccoUse/zibTobaccoUse';
export * from './zibTreatmentDirective/zibTreatmentDirective';
export * from './zibVaccination/zibVaccination';
export * from './zibVaccinationRecommendation/zibVaccinationRecommendation';

export * from './iheMhdMinimalDocumentReference/iheMhdMinimalDocumentReference';
