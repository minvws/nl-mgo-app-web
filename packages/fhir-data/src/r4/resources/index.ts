/**
 * Configurations for parsing FHIR resources and creating UI schemas.
 * Add new R4 resource configurations here.
 *
 * Notes:
 *   - Make sure you are ONLY exporting `ResourceConfig` objects from this file.
 *   - Don't forget to also export the ResourceType from the `~/api/types` file to make it available for the mobile apps.
 */

export * from './nlCoreHealthcareProvider/nlCoreHealthcareProvider';
export * from './nlCoreHealthcareProviderOrganization/nlCoreHealthcareProviderOrganization';
export * from './nlCoreHealthProfessionalPractitioner/nlCoreHealthProfessionalPractitioner';
export * from './nlCoreHealthProfessionalPractitionerRole/nlCoreHealthProfessionalPractitionerRole';
export * from './nlCorePatient/nlCorePatient';
export * from './nlCorePharmaceuticalProduct/nlCorePharmaceuticalProduct';
export * from './nlCoreVaccinationEvent/nlCoreVaccinationEvent';
