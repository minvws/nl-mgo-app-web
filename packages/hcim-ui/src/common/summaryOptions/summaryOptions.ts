import { MgoResourceMeta } from '@minvws/mgo-hcim-parse';
import { type FhirMessagesIds } from '@minvws/mgo-intl';
import { type UiContext } from '../../context/index.js';
import { type HealthUiGroup } from '../../types/index.js';

type ShowDetailsLabel =
    Extract<
        FhirMessagesIds,
        `summary.${string}.show_details`
    > extends `summary.${infer R}.show_details`
        ? R
        : never;

export function summaryOptions(
    { formatMessage }: UiContext,
    i18n: ShowDetailsLabel,
    resource: MgoResourceMeta
): HealthUiGroup {
    return {
        label: formatMessage(`summary.options`),
        children: [
            {
                type: 'REFERENCE_LINK',
                label: formatMessage(`summary.${i18n}.show_details`),
                reference: resource.referenceId,
            },
        ],
    };
}
