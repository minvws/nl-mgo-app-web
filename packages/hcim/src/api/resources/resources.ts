import { type FhirR3R4, type FhirVersion } from '@minvws/mgo-fhir';
import * as resourcesR3 from '../../r3/resources/index.js';
import * as resourcesR4 from '../../r4/resources/index.js';
import { type ResourceConfig } from '../../resourceTypes.js';

// Ensures only resource configs are listed
resourcesR3 satisfies Record<string, ResourceConfig<FhirVersion.R3, any, any>>; // eslint-disable-line @typescript-eslint/no-explicit-any
resourcesR4 satisfies Record<string, ResourceConfig<FhirVersion.R4, any, any>>; // eslint-disable-line @typescript-eslint/no-explicit-any

export const resourcesMapR3 = Object.fromEntries(
    Object.entries(resourcesR3).map(([_name, config]) => [config.profile.toLowerCase(), config])
);
export const resourcesMapR4 = Object.fromEntries(
    Object.entries(resourcesR4).map(([_name, config]) => [config.profile.toLowerCase(), config])
);

type ResourcesMapR3 = typeof resourcesMapR3;
type ResourcesMapR4 = typeof resourcesMapR4;

export type ResourceTypeConfigR3 = ResourcesMapR3[keyof ResourcesMapR3];
export type ResourceTypeConfigR4 = ResourcesMapR4[keyof ResourcesMapR4];

type MgoResourceR3 = ReturnType<ResourceTypeConfigR3['parse']>;
type MgoResourceR4 = ReturnType<ResourceTypeConfigR4['parse']>;

export type MgoResource<V extends FhirVersion | `${FhirVersion}` = FhirVersion> = FhirR3R4<
    V,
    MgoResourceR3,
    MgoResourceR4
>;
