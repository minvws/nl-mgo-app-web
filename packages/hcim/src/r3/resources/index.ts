/**
 * Configurations for parsing FHIR resources and creating UI schemas.
 * Add new R3 resource configurations here.
 *
 * Notes:
 *   - Make sure you are ONLY exporting `ResourceConfig` objects from this file.
 *   - Don't forget to also export the ResourceType from the `~/api/types` file to make it available for the mobile apps.
 */

export * from './gpDiagnosticResult/gpDiagnosticResult.js';
export * from './gpEncounter/gpEncounter.js';
export * from './gpEncounterReport/gpEncounterReport.js';
export * from './gpJournalEntry/gpJournalEntry.js';
export * from './gpLaboratoryResult/gpLaboratoryResult.js';

export * from './nlCoreEpisodeofcare/nlCoreEpisodeofcare.js';
export * from './nlCoreOrganization/nlCoreOrganization.js';
export * from './nlCorePatient/nlCorePatient.js';
export * from './nlCorePractitioner/nlCorePractitioner.js';
export * from './nlCorePractitionerRole/nlCorePractitionerRole.js';

export * from './eAfspraakAppointment/eAfspraakAppointment.js';
export {
    nlCoreObservation,
    type NlCoreObservation,
} from './nlCoreObservation/nlCoreObservation.js';
export * from './zibAdministrationAgreement/zibAdministrationAgreement.js';
export * from './zibAdvanceDirective/zibAdvanceDirective.js';
export * from './zibAlcoholUse/zibAlcoholUse.js';
export * from './zibAlert/zibAlert.js';
export * from './zibAllergyIntolerance/zibAllergyIntolerance.js';
export * from './zibBloodPressure/zibBloodPressure.js';
export * from './zibBodyHeight/zibBodyHeight.js';
export * from './zibBodyWeight/zibBodyWeight.js';
export * from './zibDrugUse/zibDrugUse.js';
export { zibEncounter, type ZibEncounter } from './zibEncounter/zibEncounter.js';
export * from './zibFunctionalOrMentalStatus/zibFunctionalOrMentalStatus.js';
export {
    zibGeneralMeasurement,
    type ZibGeneralMeasurement,
} from './zibGeneralMeasurement/zibGeneralMeasurement.js';
export {
    zibLaboratoryTestResultObservation,
    type ZibLaboratoryTestResultObservation,
} from './zibLaboratoryTestResultObservation/zibLaboratoryTestResultObservation.js';
export * from './zibLaboratoryTestResultSpecimen/zibLaboratoryTestResultSpecimen.js';
export * from './zibLaboratoryTestResultSpecimenIsolate/zibLaboratoryTestResultSpecimenIsolate.js';
export * from './zibLaboratoryTestResultSubstance/zibLaboratoryTestResultSubstance.js';
export * from './zibLivingSituation/zibLivingSituation.js';
export * from './zibMedicalDevice/zibMedicalDevice.js';
export * from './zibMedicalDeviceProduct/zibMedicalDeviceProduct.js';
export * from './zibMedicalDeviceRequest/zibMedicalDeviceRequest.js';
export * from './zibMedicationAgreement/zibMedicationAgreement.js';
export * from './zibMedicationUse/zibMedicationUse.js';
export * from './zibNutritionAdvice/zibNutritionAdvice.js';
export * from './zibPayer/zibPayer.js';
export * from './zibProblem/zibProblem.js';
export * from './zibProcedure/zibProcedure.js';
export * from './zibProcedureRequest/zibProcedureRequest.js';
export * from './zibProduct/zibProduct.js';
export * from './zibTobaccoUse/zibTobaccoUse.js';
export * from './zibTreatmentDirective/zibTreatmentDirective.js';
export * from './zibVaccination/zibVaccination.js';
export * from './zibVaccinationRecommendation/zibVaccinationRecommendation.js';

export * from './iheMhdMinimalDocumentReference/iheMhdMinimalDocumentReference.js';
