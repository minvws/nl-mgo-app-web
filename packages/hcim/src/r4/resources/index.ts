/**
 * Configurations for parsing FHIR resources and creating UI schemas.
 * Add new R4 resource configurations here.
 *
 * Notes:
 *   - Make sure you are ONLY exporting `ResourceConfig` objects from this file.
 *   - Don't forget to also export the ResourceType from the `~/api/types` file to make it available for the mobile apps.
 */

export * from './nlCoreHealthcareProvider/nlCoreHealthcareProvider.js';
export * from './nlCoreHealthcareProviderOrganization/nlCoreHealthcareProviderOrganization.js';
export * from './nlCoreHealthProfessionalPractitioner/nlCoreHealthProfessionalPractitioner.js';
export * from './nlCoreHealthProfessionalPractitionerRole/nlCoreHealthProfessionalPractitionerRole.js';
export * from './nlCorePatient/nlCorePatient.js';
export * from './nlCorePharmaceuticalProduct/nlCorePharmaceuticalProduct.js';
export * from './nlCoreVaccinationEvent/nlCoreVaccinationEvent.js';
