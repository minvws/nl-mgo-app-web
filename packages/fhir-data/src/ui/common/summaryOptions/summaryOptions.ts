import { type FhirMessagesIds } from '@minvws/mgo-mgo-intl';
import { type MgoResource } from '../../../api/resources/resources';
import { type UiHelperContext } from '../../context';
import { type HealthUiGroup } from '../../types';

type ShowDetailsLabel =
    Extract<
        FhirMessagesIds,
        `summary.${string}.show_details`
    > extends `summary.${infer R}.show_details`
        ? R
        : never;

export function summaryOptions(
    { formatMessage }: UiHelperContext,
    i18n: ShowDetailsLabel,
    resource: MgoResource
): HealthUiGroup {
    return {
        label: formatMessage(`summary.options`),
        children: [
            {
                type: 'REFERENCE_LINK',
                label: formatMessage(`summary.${i18n}.show_details`),
                reference: resource.referenceId,
            },
        ],
    };
}
