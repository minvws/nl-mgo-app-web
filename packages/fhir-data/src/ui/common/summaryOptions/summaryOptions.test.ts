import { faker } from '$test';
import { type FhirVersion } from '@minvws/mgo-fhir-types';
import { expect, test } from 'vitest';
import { type MgoResource } from '../../../api/resources/resources';
import { summaryOptions } from './summaryOptions';

test('creates a options group for the summary ui schema', () => {
    const uiHelperContext = faker.custom.uiHelperContext();
    const resource = {
        referenceId: 'MedicationStatement/zib-medicationuse-01',
    } as unknown as MgoResource<FhirVersion.R3>;
    const summaryOptionsGroup = summaryOptions(uiHelperContext, 'r3.zib_medication_use', resource);

    expect(summaryOptionsGroup).toMatchSnapshot();
});
