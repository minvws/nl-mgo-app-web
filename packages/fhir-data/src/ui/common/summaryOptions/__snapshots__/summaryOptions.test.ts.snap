// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`creates a options group for the summary ui schema 1`] = `
{
  "children": [
    {
      "label": "intl(summary.r3.zib_medication_use.show_details)",
      "reference": "MedicationStatement/zib-medicationuse-01",
      "type": "REFERENCE_LINK",
    },
  ],
  "label": "intl(summary.options)",
}
`;
